snuba/snuba_migrations/metrics/0035_metrics_sets_timeseries_id.py

```diff
@@ -0,0 +1,48 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+class Migration(migration.ClickhouseNodeMigrationLegacy):
+    """
+    Adds the timeseries_id column to the metrics sets table so we can add
+    a sharding key for each timeseries and scale the release health cluster.
+    """
+
+    blocking = False
+
+    def __forward_migrations(
+        self, table_name: str
+    ) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.METRICS,
+                table_name=table_name,
+                column=Column("timeseries_id", UInt(32)),
+            ),
+        ]
+
+    def __backward_migrations(
+        self, table_name: str
+    ) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.DropColumn(
+                storage_set=StorageSetKey.METRICS,
+                table_name=table_name,
+                column_name="timeseries_id",
+            ),
+        ]
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return self.__forward_migrations("metrics_sets_v2_local")
+
+    def backwards_local(self) -> Sequence[operations.SqlOperation]:
+        return self.__backward_migrations("metrics_sets_v2_local")
+
+    def forwards_dist(self) -> Sequence[operations.SqlOperation]:
+        return self.__forward_migrations("metrics_sets_v2_dist")
+
+    def backwards_dist(self) -> Sequence[operations.SqlOperation]:
+        return self.__backward_migrations("metrics_sets_v2_dist")
```

Comment:
I don't think this is the right table. 
(same for the tables below)

When you write on metrics, you write on a raw table and then a number of materialized views fill in the aggregate tables: `metrics_sets`, `metrics_distributions`, `metrics_counters`. 
Sharding happens on the raw table, not on the aggregate tables as the aggregate tables are colocated with the raw table. 

So this field should be added to the raw polymorphic table instead than the aggregate ones. 

This is the right table 
https://github.com/getsentry/snuba/blob/master/snuba/snuba_migrations/metrics/0020_polymorphic_buckets_table.py

Reply:
Thank you for pointing this out, I'll look into it and make the change